go/protocols/materialize/transactor.go

```diff
@@ -132,15 +163,44 @@ func RunTransactions(
 		return err
 	}
 
+	var ackDelay time.Duration
+	if d, ok := transactor.(DelayedCommitter); ok {
+		ackDelay = d.AckDelay()
+		if ackDelay > 0 {
+			log.WithField("delay", ackDelay.String()).Info("transactor running with ackDelay")
+		}
+	}
+
 	var (
 		// awaitErr is the last await() result,
 		// and is readable upon its close of its parameter `awaitDoneCh`.
 		awaitErr error
 		// loadErr is the last loadAll() result,
 		// and is readable upon its close of its parameter `loadDoneCh`.
 		loadErr error
+		// The variables used for controlling a delay on sending transaction acknowledgements to the
+		// runtime. This will "spread out" transaction processing and result in fewer, larger
+		// transactions which may be desirable to reduce warehouse compute costs or comply with rate
+		// limits. They are safe to access in the concurrent await function because they are only
+		// set in the "round" loop after awaitDoneCh has closed.
+		//
+		// commitStartedAt is the time at which the the commit for the latest round began.
+		commitStartedAt time.Time
+		// storedHistory is a list of the most recent stored counts for transactions.
+		storedHistory [storedHistorySize]int
 	)
 
+	// True only if all historical stored counts are below the threshold, and none of them are 0.
+	// Counts at 0 means that many transactions haven't been completed yet.
```

Comment:
The zero value is strange. Can we not wait until `startCommit` is called and finished before we store `it.Total` in `storedHistory`?